### PR TITLE
[admin] Added monitoring status graph on dashboard

### PR DESCRIPTION
**Pull request by pandafy**
Checks:

- [x] I have manually tested the proposed changes
- [ ] I have written new test cases to avoid regressions (if necessary)
- [ ] ~I have updated the documentation (e.g. README.rst)~

--------
image
--------

Along with https://github.com/openwisp/openwisp-controller/pull/369, it looks like below 
Screenshot from 2021-01-16 17-17-57

--------

**Blockers**:

- [ ] https://github.com/openwisp/openwisp-utils/pull/168
- [ ] https://github.com/openwisp/openwisp-controller/pull/369

### CODE REVIEW

**Review by nemesisdesign — COMMENTED** [review]
This is what I see:

Screenshot from 2021-01-19 22-31-37

The labels of the monitoring status looks cut.

**Comment by pandafy** [comment]
@nemesisdesign  I spent some time trying to debug this but couldn't find the reason why this was happening. I noticed that it only happened in Chromium when testing with responsive toolbar. I tested it with different view port size by resizing the browser window and it worked fine. 

It also worked fine when tested with responsive toolbar of Firefox.

Were you testing it in responsive toolbar of Chrome or Chromium?  

I saw [many posts on the community forum of plotly](https://community.plotly.com/search?q=legend%20truncated) complaining about the same issue with horizontal legend. If you are facing this problem in normal browser view then we might have to switch to vertical legend.

**Comment by nemesisdesign** [comment]
> @nemesisdesign I spent some time trying to debug this but couldn't find the reason why this was happening. I noticed that it only happened in Chromium when testing with responsive toolbar. I tested it with different view port size by resizing the browser window and it worked fine.
> 
> It also worked fine when tested with responsive toolbar of Firefox.
> 
> Were you testing it in responsive toolbar of Chrome or Chromium?
> 
> I saw [many posts on the community forum of plotly](https://community.plotly.com/search?q=legend%20truncated) complaining about the same issue with horizontal legend. If you are facing this problem in normal browser view then we might have to switch to vertical legend.

I tested 

> @nemesisdesign I spent some time trying to debug this but couldn't find the reason why this was happening. I noticed that it only happened in Chromium when testing with responsive toolbar. I tested it with different view port size by resizing the browser window and it worked fine.
> 
> It also worked fine when tested with responsive toolbar of Firefox.
> 
> Were you testing it in responsive toolbar of Chrome or Chromium?
> 
> I saw [many posts on the community forum of plotly](https://community.plotly.com/search?q=legend%20truncated) complaining about the same issue with horizontal legend. If you are facing this problem in normal browser view then we might have to switch to vertical legend.

It happens also on Firefox to me. It happens only in the monitoring pie chart, not in the controller chart.

**Comment by pandafy** [comment]
image

I have updated the orientation of the legend to vertical instead of horizontal

@nemesisdesign where to define custom dependencies, since `./install-dev.sh` is not present anymore.

**Comment by nemesisdesign** [comment]
> image
> 
> I have updated the orientation of the legend to vertical instead of horizontal

Thx, will review and test asap.

> @nemesisdesign where to define custom dependencies, since `./install-dev.sh` is not present anymore.

I recently (finally!) found a better way of managing these dev dependencies, see:
https://github.com/openwisp/openwisp-radius/blob/d11d8bdba0e1519697cd78b5aa97a42f6aca79ac/requirements.txt#L7-L9

Use this method from now in all the repos please, for more info see [PEP 508](https://www.python.org/dev/peps/pep-0508/).

File: openwisp_monitoring/monitoring/apps.py [hunk 1]
```diff
@@ -23,6 +25,24 @@ def ready(self):
         for metric_name, metric_config in metrics.items():
             register_metric_notifications(metric_name, metric_config)
 
+        register_dashboard_element(
```
Comment (nemesisdesign) [inline openwisp_monitoring/monitoring/apps.py:28]: this code should go in `openwisp_monitoring.device.apps`

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

**Comment by pandafy** [comment]
I can't understand the reason why pip is not installing the latest version of openwisp-utils. I even enable `-I` in the installation command to ignore installed packages, but it has no effect. I will get back to it tomorrow.

File: openwisp_monitoring/device/api/views.py [hunk 2]
```diff
@@ -357,3 +367,66 @@ def _create_resources_alert_settings(self, metric, resource):
...
+
+
+class MonitoringDeviceSerializer(LocationDeviceSerializer):
+    monitoring = DeviceMonitoringSerializer()
+
+
+class MonitoringLocationDeviceList(LocationDeviceList):
+    serializer_class = MonitoringDeviceSerializer
+
+    # def get(self, *args, **kwargs):
```
Comment (nemesisdesign) [inline openwisp_monitoring/device/api/views.py:423]: TODO: remove this

File: openwisp_monitoring/device/static/monitoring/js/device-map.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      });
+      el.find('.prev').click(function(){
+        loadPopUpContent(layer, $(this).data('url'));
+      });
+
+      loadingOverlay.hide();
+
+    }).fail(function(){
+      loadingOverlay.hide();
+      alert('Error while retrieving data');
```
Comment (nemesisdesign) [inline openwisp_monitoring/device/static/monitoring/js/device-map.js:117]: TODO: use `gettext()` here

File: openwisp_monitoring/device/tests/test_api.py [hunk 4]
```diff
@@ -531,3 +531,9 @@ def test_garbage_mobile_properties(self):
             number += 1
             with self.subTest(interface_data['name']):
                 self.assertEqual(r.status_code, 400)
+
+    def test_geojson_api(self):
+        pass
+        # device = self._create_device(organization=self._create_org())
+        # TODO ...
+        # Location()
```
Comment (nemesisdesign) [inline openwisp_monitoring/device/tests/test_api.py:539]: TODO add basic API tests

File: tests/openwisp2/settings.py [hunk 5]
```diff
@@ -144,6 +144,7 @@
 EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
 
 OPENWISP_MONITORING_MANAGEMENT_IP_ONLY = False
+OPENWISP_ADMIN_DASHBOARD_ENABLED = True
```
Comment (nemesisdesign) [inline tests/openwisp2/settings.py:147]: TODO: remove this, shouldn't be needed

**Comment by nemesisdesign** [comment]
This is how it looks:

dashboard1

**Comment by nemesisdesign** [comment]
dashboard2

**Pull request merged**